src/lib.rs

```diff
@@ -0,0 +1,571 @@
+        &DEFAULT_COMPUTE_HEAVY_FUTURE_EXECUTOR_STRATEGY.get_or_init(|| {
+            #[cfg(feature = "tokio")]
+            {
+                log::info!("Defaulting to SpawnBlocking strategy for compute-heavy future executor \
+                until a strategy is initialized");
+
+                ExecutorStrategyImpl::SpawnBlocking(SpawnBlockingExecutor::new(None))
+            }
+
+            #[cfg(not(feature = "tokio"))]
+            {
+                log::warn!("Defaulting to CurrentContext (non-op) strategy for compute-heavy future executor \
+                until a strategy is initialized.");
+                ExecutorStrategyImpl::CurrentContext(CurrentContextExecutor::new(None))
+            }
+        })
+    }
+}
+
+/// Spawn a future to the configured compute-heavy executor and wait on its output.
+///
+/// # Strategy selection
+///
+/// If no strategy is configured, this library will fall back to the following defaults:
+/// - no `tokio`` feature - current context
+/// - all other cases - spawn blocking
+///
+/// You can override these defaults by initializing a strategy via [`global_strategy_builder()`]
+/// and [`GlobalStrategyBuilder`].
+///
+/// # Cancellation
+///
+/// Most strategies will cancel the input future, if the caller drops the returned future,
+/// with the following exception:
+/// - the block in place strategy never cancels the future (until the executor is shut down)
+///
+/// # Example
+///
+/// ```
+/// # async fn run() {
+/// use compute_heavy_future_executor::run_compute_heavy_future;
+///
+/// let future = async {
+///     std::thread::sleep(std::time::Duration::from_millis(50));
+///     5
+///  };
+///
+/// let res = run_compute_heavy_future(future).await.unwrap();
+/// assert_eq!(res, 5);
+/// # }
+///
+/// ```
+///
+pub async fn run_compute_heavy_future<F, R>(fut: F) -> Result<R, Error>
+where
+    F: Future<Output = R> + Send + 'static,
+    R: Send + 'static,
+{
+    let executor = COMPUTE_HEAVY_FUTURE_EXECUTOR_STRATEGY
+        .get()
+        .unwrap_or_else(|| <&ExecutorStrategyImpl>::default());
+    match executor {
+        ExecutorStrategyImpl::CurrentContext(executor) => executor.execute(fut).await,
+        ExecutorStrategyImpl::CustomExecutor(executor) => executor.execute(fut).await,
+        #[cfg(feature = "tokio_block_in_place")]
+        ExecutorStrategyImpl::BlockInPlace(executor) => executor.execute(fut).await,
+        #[cfg(feature = "tokio")]
+        ExecutorStrategyImpl::SpawnBlocking(executor) => executor.execute(fut).await,
+        #[cfg(feature = "secondary_tokio_runtime")]
+        ExecutorStrategyImpl::SecondaryTokioRuntime(executor) => executor.execute(fut).await,
+    }
+}
+
+pub fn make_future_cancellable<F, O>(fut: F) -> (impl Future<Output = ()>, Receiver<O>)
+where
+    F: std::future::Future<Output = O> + Send + 'static,
+    O: Send + 'static,
+{
+    let (mut tx, rx) = tokio::sync::oneshot::channel();
+    let wrapped_future = async {
+        select! {
```

Comment:
should you use a weighed rather than a random select here?

Reply:
Thanks, I was thinking it was biased by default.